mailbox/plugin/deleted-messages-vault/src/main/java/org/apache/james/vault/DeletedMessageConverter.java

```diff
@@ -0,0 +1,139 @@
+
+package org.apache.james.vault;
+
+import java.io.IOException;
+import java.time.Clock;
+import java.time.ZoneOffset;
+import java.time.ZonedDateTime;
+import java.util.Date;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.stream.Stream;
+
+import javax.mail.internet.AddressException;
+
+import org.apache.james.core.MailAddress;
+import org.apache.james.core.MaybeSender;
+import org.apache.james.core.User;
+import org.apache.james.mailbox.store.mail.model.MailboxMessage;
+import org.apache.james.mime4j.MimeIOException;
+import org.apache.james.mime4j.codec.DecodeMonitor;
+import org.apache.james.mime4j.dom.Message;
+import org.apache.james.mime4j.dom.address.AddressList;
+import org.apache.james.mime4j.dom.address.Mailbox;
+import org.apache.james.mime4j.dom.address.MailboxList;
+import org.apache.james.mime4j.message.DefaultMessageBuilder;
+import org.apache.james.mime4j.stream.MimeConfig;
+import org.apache.james.util.StreamUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.github.steveash.guavate.Guavate;
+import com.google.common.base.Preconditions;
+
+class DeletedMessageConverter {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(DeletedMessageConverter.class);
+
+    private final Clock clock;
+
+    DeletedMessageConverter(Clock clock) {
+        this.clock = clock;
+    }
+
+    DeletedMessage convert(DeletedMessageMetadata deletedMessageMetadata, MailboxMessage mailboxMessage) throws IOException {
+        Preconditions.checkNotNull(deletedMessageMetadata);
+        Preconditions.checkNotNull(mailboxMessage);
+
+        Optional<Message> mimeMessage = parseMessage(mailboxMessage);
+
+        return DeletedMessage.builder()
+            .messageId(deletedMessageMetadata.getMessageId())
+            .originMailboxes(deletedMessageMetadata.getOwnerMailboxes())
+            .user(retrieveOwner(deletedMessageMetadata))
+            .deliveryDate(retrieveDeliveryDate(mimeMessage, mailboxMessage))
+            .deletionDate(ZonedDateTime.ofInstant(clock.instant(), ZoneOffset.UTC))
+            .sender(retrieveSender(mimeMessage))
+            .recipients(retrieveRecipients(mimeMessage))
+            .hasAttachment(mailboxMessage.getAttachments().iterator().hasNext())
+            .subject(mimeMessage.map(Message::getSubject))
+            .build();
+    }
+
+    private Optional<Message> parseMessage(MailboxMessage mailboxMessage) throws IOException {
+        DefaultMessageBuilder messageBuilder = new DefaultMessageBuilder();
+        messageBuilder.setMimeEntityConfig(MimeConfig.PERMISSIVE);
+        messageBuilder.setDecodeMonitor(DecodeMonitor.SILENT);
+        try {
+            return Optional.ofNullable(messageBuilder.parseMessage(mailboxMessage.getFullContent()));
+        } catch (MimeIOException e) {
+            LOGGER.warn("Can not parse the message {}", mailboxMessage.getUid(), e);
```

Comment:
UID makes sense in a mailbox context, but not by its own.

Please rely on 'MessageId'